### PR TITLE
Reflect json path in error message

### PR DESCRIPTION
**Pull request by nojaf**
### Related Issues/PRs

 #12550

### What changes are proposed in this pull request?

This is an attempt to have more consistent error messages. Tackling the `_log_batch` endpoint specifically. (To start somewhere).

Proposal would be to settle on the following template:

```
Invalid value {value} for parameter 'path' supplied: {extra}
```

Where:

- `value` is processed by `json.dumps` and trimmed if too long.
- `path` represent the json path of the current value.
- `extra` is additional information about what actually went wrong.

Example:

```
Invalid value \"yxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx for parameter 'params[0].value' supplied"
```

This is just a proposal, happy to hear your thoughts on this!

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Slightly different error messages in API responses.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

Maybe, I'm not quite sure here.

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/utils/validation.py [hunk 1]
```diff
@@ -96,6 +98,25 @@
...
+
+
+def append_to_json_path(currenPath, value):
+    if not currenPath:
+        return value
+
+    if value.startswith("["):
+        return f"{currenPath}{value}"
+
+    return f"{currenPath}.{value}"
```
Comment (BenWilson2) [inline mlflow/utils/validation.py:117]:
```suggestion
def append_to_json_path(currentPath, value):
    if not currentPath:
        return value

    if value.startswith("["):
        return f"{currentPath}{value}"

    return f"{currentPath}.{value}"
```

**Comment by nojaf** [comment]
@BenWilson2 I think this is a good first step and improves the messages in `log_batch`.
Would like to improve more endpoints in subsequent PRs.

**Comment by nojaf** [comment]
I've been using these changes in https://github.com/jgiannuzzi/mlflow-go/pull/24 and I'm able to streamline both the Python and Go error responses.

**Review by BenWilson2 — APPROVED** [review]
Positive changes here! I like these more standard exception messages :)

**Comment by BenWilson2** [comment]
There will be a handful of test-only failures to clean up to get this merged. Let me know when it's all set and we'll get this merged @nojaf

**Comment by nojaf** [comment]
Thanks @BenWilson2, my main problem is that I cannot just run `pytest tests` in the dev container.
So without CI, it is difficult to spot which tests aren't passing anymore.

I've updated the failing tests and this is ready for another go!

**Pull request merged**